### DISCUSSION TITLE
Wiring Diagram Hyperhdr hyperspi on s2 mini with level shifter

### DISCUSSION THREAD

**Discussion by Todderbod** [awawa-dev/HyperHDR#559] [Q&A]
I have searched and can not find a diagram of how to wire Hyperhdr with esp32 s-2 mini running hyperspi with a level shifter. Does there need to be power ran to the s2 mini or is it powered by the spi.

**Comment by awawa-dev** [comment]
Power supply for s2 mini must be provided via USB port, just like for esp32.

**Reply by Todderbod** [reply]
Thank you so much I appreciate the quick response

**Comment by AstaRom** [comment]
> how to wire Hyperhdr with esp32 s-2 mini running hyperspi with a level shifter.

![ESP32-S2 mini](https://user-images.githubusercontent.com/39379875/225410503-dfa575e0-2ecf-433a-98ac-258e47f11b2e.png)

**Comment by jt7005** [comment]
What pins are used if using a Raspberry Pi 3 Model B?

**Reply by awawa-dev** [reply]
The same
